src/twisted/internet/testing.py

```diff
@@ -414,8 +420,44 @@ def getDestination(self):
         return self._address
 
 
+@implementer(IHostResolution)
+@dataclass
+class ImmediateResolution:
```

Comment:
Mabye this should have the private marker as even for testing, it should not be used outside of the `ImmediateResolver`

```suggestion
class _ImmediateResolution:
```

Reply:
Good catch, yes.